native/common/include/jp_env.h

```diff
@@ -17,27 +17,16 @@
 #ifndef _JPENV_H_
 #define _JPENV_H_
 
-/** Java method of managing local references.
- *
- * This should be used around all entry points from python that 
- * call java code.  Failure may lead to local references not being
- * released.  Methods will large numbers of local references 
- * should allocate a local frame.  At most one local reference 
- * from a local frame can be kept.  Addition must use global referencing.
+/**
+ *  the platform adapter's implementation is chosen by the JPYPE_??? macros
  */
-static const int LOCAL_FRAME_DEFAULT=8;
-class JPLocalFrame
+class JPPlatformAdapter

```

Comment:
Most of the JPJavaEnv has been folded into JPEnv and JPJavaFrame.  It is now required that you have a JPJavaFrame to access java classes so that it is impossible to create a leaking local reference or attempt to use something without an attached thread.